plugins/by-name/none-ls/prettier.nix

```diff
@@ -1,38 +1,41 @@
-{ lib, config, ... }:
+{
+  lib,
+  config,
+  options,
+  ...
+}:
 let
   cfg = config.plugins.none-ls.sources.formatting.prettier;
   ts-ls-cfg = config.plugins.lsp.servers.ts_ls;
+  opt = options.plugins.none-ls.sources.formatting.prettier;
+  defaultPrio = (lib.mkOptionDefault null).priority;
 in
 {
-  options.plugins.none-ls.sources.formatting.prettier = {
+  opt = {
     disableTsServerFormatter = lib.mkOption {
-      type = with lib.types; nullOr bool;
+      type = lib.types.bool;
```

Comment:
> I also refactored the `prettier.nix` config to use the same priority system instead of checking against `null`. This should only be an issue if somebody has explicitly set their option to `null`, otherwise it should work the same.
> 
> Let me know if you want that reversed.

If we wanted to have some kinda graceful transition period we'd have to create a type along the lines of:
```nix
# A nullOr bool that pretends it is just a bool in its description
fakeBoolType = lib.types.nullOr lib.types.bool // {
  description = lib.types.bool.description;
}
```
and also have a separate warning for when the option is null.

In this case, I don't think that's worth the effort because (as you say) no one should be explicitly setting this option to null.

If anyone is explicitly setting it to null, they should get a fairly helpful error about using an incorrect type for this option.